### ISSUE TITLE
how to fix wrong RSS USLs in the suggestions list?

### ISSUE THREAD

**Issue by TheOtherDoctor** [FredJul/Flym#561]
I have a few feed URLs identified wrong - that is, the entry resulting from the feed add/search dialog is an invalid URL giving no feed when updating, and I could manually find/guess the correct one which works; now I would also fix this upstream but I could not find where/how flym gets its feed search results. Could somebody give a hint how this is implemented and if/where I should look to fix wrong data?
This is probably documented somewhere... sorry for me failing to find. Not a coder - at least not for this environment. ;-)

**Comment by ildar** [comment]
Did you check long tap on a feed?

**Comment by TheOtherDoctor** [comment]
Yes, and sorry for obviously being unclear: I know how to fix it for me in flym, that's very intuitive. What I wonder is, if I can somehow bring my correction upstream into whatever source flym uses to search for the RSS URLs it presents; that's what I meant with the functionality which implementation I could not find in the code.

**Comment by ildar** [comment]
This app cannot fix RSSs it uses nor even notify someone about anything.
Could you explain your idea more technically?

**Comment by FredJul** [comment]
Hi,
You can find the search code into the file named FeedSearchDialog. You can see there that this is using the Feedly API to do the query (unfortunately this is not documented).

I clearly encourage external contributions, especially since I do not have much time myself to improve Flym.

**Comment by TheOtherDoctor** [comment]
Ok, then I'd likely have to dig into the feedly database to actually fix wrong info from there; will have a look if/how this is practically feasible.

For the question on what flym could do on wrong/broken URLs: currently, as far as I could observe,  a feed just "turns red" when nothing gets received. I'd suggest to, when trying to open such a feed, put a flym-generated pseudo news message describing the situation (like "feed URL did not provide any feed data" and mention that "Note that URLs suggested in search are provided by feedly and might need updating in the DB"; from my perspective, that would be already a step ahead.

A bit more proactive would be to run a "test feed" function in the search dialog, immediately after picking the suggested feed from the search (when we are still sure that the URL has not been manually edited) giving it a try if download produces a list of entries; if this fails then, a more specific hint like "the URL you just picked from the list seems incorrect, consider reporting this to feedly if the prolem persists" or the like?

**Comment by TheOtherDoctor** [comment]
I could not find an interface to suggest corrections to feedly. As for my specific issue is long solved by manual edit in flym, and if community contributions are no part of feedly's concept, I'd tend to give up on trying to fix anything actively for now - probably soon the fix will happen in feedly anyway, due to whatever update mechanism. ;-)
